### ISSUE TITLE
Issue: CacheMode Auto not working on cmder session from Windows Terminal

### ISSUE THREAD

**Issue by Lunchb0ne** [gerardog/gsudo#98]
## Steps to Reproduce

1. Set config option to set cache to auto.
2. Restart Windows Terminal
3. Try testing out sudo persistence, it asks for elevation on every command

## Screenshots

![image](https://user-images.githubusercontent.com/22198661/128665415-d4896cfb-7e99-4d5a-abd7-1d46e732ae67.png)
This window opens when I try and run sudo in debug mode

```bash
# First Run on a new terminal session
~
λ sudo --debug powercfg /?
Debug: Command to run: "C:\WINDOWS\system32\powercfg.EXE" /?
Debug: Using Console mode TokenSwitch
Debug: Caller PID: 12044
Debug: Caller SID: S-1-5-21-778842817-3590866686-2318578362-1001
Debug: Elevating process: C:\Users\abhis\scoop\apps\gsudo\1.0.0\gsudo.exe --debug gsudoservice 12044 S-1-5-21-778842817-3590866686-2318578362-1001 All 00:05:00
Debug: Elevated instance started.
Debug: Connected via Named Pipe ProtectedPrefix\Administrators\gsudo_964B7E7B288BAA5FBB7AA33D8AB31DDF781BA8CB818EA01F7C39563C51161812.
Debug: CreateProcessAsUser: "C:\WINDOWS\system32\powercfg.EXE" /?
Debug: ElevationRequest length 788
Debug: Process token successfully substituted.
```

```bash
# Second run
~
λ sudo --debug powercfg /?
Debug: Command to run: "C:\WINDOWS\system32\powercfg.EXE" /?
Debug: Using Console mode TokenSwitch
Debug: Caller PID: 24324
Debug: Caller SID: S-1-5-21-778842817-3590866686-2318578362-1001
Debug: Elevating process: C:\Users\abhis\scoop\apps\gsudo\1.0.0\gsudo.exe --debug gsudoservice 24324 S-1-5-21-778842817-3590866686-2318578362-1001 All 00:05:00
Debug: Elevated instance started.
Debug: Connected via Named Pipe ProtectedPrefix\Administrators\gsudo_875AEEDB854BAB538762027ACE034694448D4ADAF0B3632D2679DC9B7A0B7EC7.
Debug: CreateProcessAsUser: "C:\WINDOWS\system32\powercfg.EXE" /?
Debug: ElevationRequest length 788
Debug: Process token successfully substituted.
```

It seems like that the PIDs change and that doesn't let it persist in the cache.
But if I run the config command and set it to `Auto` for every new terminal session, it somehow manages to persist.
gSudo Service
![image](https://user-images.githubusercontent.com/22198661/128665863-8667df74-dfdf-4c5f-9e42-b5901a34f463.png)

```bash
~
λ gsudo --debug config cachemode auto
Info: Config Setting for 'CacheMode' will be set as global system setting.
Info: Global system settings requires elevation. Elevating...
Debug: Command to run: "C:\Users\abhis\scoop\apps\gsudo\1.0.0\gsudo.exe" --global config cachemode "auto"
Debug: Using Console mode TokenSwitch
Debug: Caller PID: 24568
Debug: Caller SID: S-1-5-21-778842817-3590866686-2318578362-1001
Debug: Elevating process: C:\Users\abhis\scoop\apps\gsudo\1.0.0\gsudo.exe --debug gsudoservice 24568 S-1-5-21-778842817-3590866686-2318578362-1001 All 00:05:00
Debug: Elevated instance started.
Debug: Connected via Named Pipe ProtectedPrefix\Administrators\gsudo_52F58BAD3703BBC7E06967397F2CC9B4D5EED15642BDA15529319719185C14CE.
Debug: CreateProcessAsUser: "C:\Users\abhis\scoop\apps\gsudo\1.0.0\gsudo.exe" --global config cachemode "auto"
Debug: ElevationRequest length 833
Debug: Process token successfully substituted.
Warning: Enabling the credentials cache is a security risk.
CacheMode = "Auto" (global)
Debug: Process exited with code 0

~
λ sudo --debug powercfg /?
Debug: Command to run: "C:\WINDOWS\system32\powercfg.EXE" /?
Debug: Using Console mode TokenSwitch
Debug: Caller PID: 21040
Debug: Found Named Pipe "ProtectedPrefix\Administrators\gsudo_52F58BAD3703BBC7E06967397F2CC9B4D5EED15642BDA15529319719185C14CE".
Debug: Connected via Named Pipe ProtectedPrefix\Administrators\gsudo_52F58BAD3703BBC7E06967397F2CC9B4D5EED15642BDA15529319719185C14CE.
Debug: CreateProcessAsUser: "C:\WINDOWS\system32\powercfg.EXE" /?
Debug: ElevationRequest length 788
Debug: Process token successfully substituted.
```
## Context:
 - Windows version: 21H2 (Build 22000.120)
 - gsudo version:
	```
	gsudo v1.0.0 (Branch.master.Sha.dccde648353a6ffade9e04e484858a15bd82e3b0)
	Copyright(c) 2019-2021 Gerardo Grignoli and GitHub contributors
	```

**Comment by gerardog** [comment]
Hi, 
I am not sure if you are describing a problem or the actual expected behavior. The readme says:

> Cache Modes:
> 
> - Auto: Simil-unix-sudo. The first elevation shows a UAC Popup and starts a cache session automatically.
> - Explicit: (default) **Every elevation shows a UAC popup**, unless a cache session is started explictly with gsudo cache on.
> - Disabled: Every elevation request shows a UAC popup.

By default the cache does not starts automatically, every elevation request shows a UAC popup. That is the intended behaviour.
But, *if* after running `gsudo config cachemode auto` only one UAC popup appears (per console/tab), then it is working as expected.

Please let me know how it works with CacheMode auto (the setting is persistent, you only need to set it once). If the UAC only appears once per console, then it is working as expected. Otherwise, please let me know. Thanks

**Comment by Lunchb0ne** [comment]
> Hi,
> I am not sure if you are describing a problem or the actual expected behavior. The readme says:
> 
> > Cache Modes:
> > 
> > * Auto: Simil-unix-sudo. The first elevation shows a UAC Popup and starts a cache session automatically.
> > * Explicit: (default) **Every elevation shows a UAC popup**, unless a cache session is started explictly with gsudo cache on.
> > * Disabled: Every elevation request shows a UAC popup.
> 
> By default the cache does not starts automatically, every elevation request shows a UAC popup. That is the intended behaviour.
> But, _if_ after running `gsudo config cachemode auto` only one UAC popup appears (per console/tab), then it is working as expected.
> 
> Please let me know how it works with CacheMode auto (the setting is persistent, you only need to set it once). If the UAC only appears once per console, then it is working as expected. Otherwise, please let me know. Thanks

Yep this is the issue, I have cache mode already set to Auto from before, but seems like Everytime I start a new terminal session it doesn't seem to Respect that information

**Comment by gerardog** [comment]
How did you installed gsudo? 
Have you tried using `gsudo` instead of `sudo`?

It works for me! I've recorded my screen using windows Game Bar (Win+G key), then reduced the video size using Handbrake to upload it here. Uploading to a third party web would also work.

https://user-images.githubusercontent.com/3901474/128906338-865e2b07-df26-4b71-9bfa-4c663ccab8ab.mp4

**Comment by gerardog** [comment]
Please note that the cache only allows elevation from one parent process, i.e. the invoker console. For example, open one console, elevate many times, with only one popup, but the moment you open another console (like a new WT tab) it will ask again, once per tab.

**Comment by Lunchb0ne** [comment]
> How did you installed gsudo?
> Have you tried using `gsudo` instead of `sudo`?
> 
> It works for me! I've recorded my screen using windows Game Bar (Win+G key), then reduced the video size using Handbrake to upload it here. Uploading to a third party web would also work.
> 
>  Cmder.2021-08-10.14-09-45-1.mp4

Seems like using `gsudo` instead of `sudo` solves the problem. My main issue was that if I open a new terminal tab, the expected behaviour was that it would UAC prompt me for the first time and then the session for the same tab should persist for some time, but it simply ended up breaking.

Here's an example:
The blacking out is because of the invocation of the UAC prompt.
https://user-images.githubusercontent.com/22198661/128978526-92703b92-1fe5-4c0d-9d9f-9a10067f0c14.mp4

**Comment by gerardog** [comment]
How did you installed gsudo?
What is the output of `where gsudo` on Cmder?

Please do this test: run `sudo status` and paste here the results. 
Thanks!

**Comment by Lunchb0ne** [comment]
I installed gsudo via [scoop](https://github.com/lukesampson/scoop).

Output for `where gsudo`
```bash
C:\Users\abhis\scoop\apps\gsudo\current\gsudo
C:\Users\abhis\scoop\apps\gsudo\current\gsudo.exe
```

Output for `sudo status`
```bash
Caller Pid: 22756
Running as:
  User: VALKYRIE-HOME\abhis
  Sid: S-1-5-21-778842817-3590866686-2318578362-1001
  Is Admin: False
  Integrity Level: Medium (8192)

Credentials Cache:
  Mode: Auto
  Available for this process: False
  Total active cache sessions: 0

Processes attached to the current console:
      PID Integrity  UserName                  Name
    17952 Medium     VALKYRIE-HOME\abhis       C:\WINDOWS\system32\cmd.exe
    22756 Medium     VALKYRIE-HOME\abhis       C:\Users\abhis\scoop\shims\sudo.exe
    17024 Medium     VALKYRIE-HOME\abhis       C:\Users\abhis\scoop\apps\gsudo\current\gsudo.exe (this gsudo status)
```

**Comment by gerardog** [comment]
Fixed in v1.0.2
